packages/react-pages/src/node/index.ts

```diff
@@ -54,9 +56,24 @@ export default function pluginFactory(
         },
       },
     }),
-    configResolved(config) {
-      pagesDir = opts.pagesDir ?? path.resolve(config.root, 'pages')
+    configResolved({ root, plugins, logger }) {
+      pagesDir = opts.pagesDir ?? path.resolve(root, 'pages')
       pageStrategy = new PageStrategy(pagesDir, findPages, loadPageData)
+
+      // Inject parsing logic for frontmatter if missing.
+      if (!resolve.silent(root, 'remark-frontmatter')) {
+        const mdxPlugin = plugins.find(
+          (plugin) => plugin.name === 'vite-plugin-mdx'
+        ) as MdxPlugin | undefined
+
+        if (mdxPlugin?.mdxOptions) {
+          mdxPlugin.mdxOptions.remarkPlugins.push(require('remark-frontmatter'))
```

Comment:
I the code run to here, it means the user project doesn't have `remark-frontmatter` installed. So I think this `require('remark-frontmatter')` must throw error.

Reply:
Oh, this *does* introduce an issue for Yarn users, since flat node_modules leads to an incorrect detection of `remark-frontmatter` in the project.

I think we need to check `package.json` of root.